Sources/SwiftParser/IsValidIdentifier.swift

```diff
@@ -0,0 +1,70 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+/// Context in which to check if a name can be used as an identifier.
+///
+/// - SeeAlso: ``isValidIdentifier``
+public enum IdentifierCheckContext {
+  /// Check if a name can be used to declare a variable, ie. if it can be used after a `let` or `var` keyword.
+  ///
+  /// ### Examples
+  /// - `test` is a valid variable name and `let test: Int` is valid Swift code
+  /// - `class` is not a valid variable and `let class: Int` is invalid Swift code
+  case variableName
+}
+
+/// Checks whether `name` can be used as an identifier in a certain context.
+///
+/// If the name cannot be used as an identifier in this context, it needs to be escaped.
+/// For example, `class` is not a valid identifier for a variable name and needs to be spelled as ```let `class`: String``` to be valid Swift code.
+///
+/// The context is important here some names can be used as identifiers in some contexts but not others. For example, `myStruct.class` is valid without adding backticks `class`, but as mentioned above, backticks need to be added when `class` is used as a variable name.
```

Comment:
```suggestion
/// The context is important here some names - can be used as identifiers in some contexts but not others. For example, `myStruct.class` is valid without adding backticks `class`, but as mentioned above, backticks need to be added when `class` is used as a variable name.
```

It's a little odd that we mention the `myStruct.class` case but don't have a corresponding case for it in `IdentifierCheckContext`.

Reply:
OK, added a new case to also check if a name is valid identifier for member access. Seems reasonable to have that as well anyway.